src/irr/asset/CGLILoader.cpp

```diff
@@ -48,284 +48,385 @@ namespace irr
 
-			return SAssetBundle({ core::smart_refctd_ptr<IAsset>(rawImagesBundle, core::dont_grab) });
+			//return SAssetBundle( ? );
 		}
 
 		bool CGLILoader::isALoadableFileFormat(io::IReadFile* _file) const
 		{
 			return true; // gli provides a function to load files, but we can check files' signature actually if needed
 		}
 
-		inline E_FORMAT CGLILoader::getTranslatedGLIFormat(const gli::texture& texture, const gli::gl& glVersion)
+		inline std::pair<E_FORMAT, ICPUImageView::SComponentMapping> CGLILoader::getTranslatedGLIFormat(const gli::texture& texture, const gli::gl& glVersion)
 		{
 			using namespace gli;
 			gli::gl::format formatToTranslate = glVersion.translate(texture.format(), texture.swizzles());
+			ICPUImageView::SComponentMapping compomentMapping;
+
+			constexpr std::array<std::pair<gli::gl::swizzle, ICPUImageView::SComponentMapping::E_SWIZZLE>, 6> swizzlesMappingAPI =
+			{
+				std::make_pair(gl::SWIZZLE_RED, ICPUImageView::SComponentMapping::ES_R),
+				std::make_pair(gl::SWIZZLE_GREEN, ICPUImageView::SComponentMapping::ES_G),
+				std::make_pair(gl::SWIZZLE_BLUE, ICPUImageView::SComponentMapping::ES_B),
+				std::make_pair(gl::SWIZZLE_ALPHA, ICPUImageView::SComponentMapping::ES_A),
+				std::make_pair(gl::SWIZZLE_ONE, ICPUImageView::SComponentMapping::ES_ONE),
+				std::make_pair(gl::SWIZZLE_ZERO, ICPUImageView::SComponentMapping::ES_ZERO)
+			};
+
+			auto getMappedSwizzle = [&](const gli::gl::swizzle& currentSwizzleToCheck)
+			{
+				for (auto& mappedSwizzle : swizzlesMappingAPI)
+					if (currentSwizzleToCheck == mappedSwizzle.first)
+						return mappedSwizzle.second;
+			};
+
+			compomentMapping.r = getMappedSwizzle(static_cast<gli::gl::swizzle>(formatToTranslate.Swizzles.r));
+			compomentMapping.g = getMappedSwizzle(static_cast<gli::gl::swizzle>(formatToTranslate.Swizzles.g));
+			compomentMapping.b = getMappedSwizzle(static_cast<gli::gl::swizzle>(formatToTranslate.Swizzles.b));
+			compomentMapping.a = getMappedSwizzle(static_cast<gli::gl::swizzle>(formatToTranslate.Swizzles.a));
+
+			auto getTranslatedFinalFormat = [&](const E_FORMAT& format = EF_UNKNOWN)
+			{
+				return std::make_pair(format, compomentMapping);
+			};
 
-			// TODO
+			// TODO - fill formats
 			switch (formatToTranslate.Internal)
 			{
-				case gl::INTERNAL_RGB_UNORM: return EF_R8G8B8_UNORM;			//GL_RGB
-				case gl::INTERNAL_BGR_UNORM: return;	//GL_BGR
-				case gl::INTERNAL_RGBA_UNORM: return;		//GL_RGBA
-				case gl::INTERNAL_BGRA_UNORM: return;		//GL_BGRA
-				case gl::INTERNAL_BGRA8_UNORM: return;		//GL_BGRA8_EXT
+				case gl::INTERNAL_RGB_UNORM: return getTranslatedFinalFormat(EF_R8G8B8_UNORM);			//GL_RGB
+				case gl::INTERNAL_BGR_UNORM: return getTranslatedFinalFormat();			//GL_BGR
+				case gl::INTERNAL_RGBA_UNORM: return getTranslatedFinalFormat();		//GL_RGBA
+				case gl::INTERNAL_BGRA_UNORM: return getTranslatedFinalFormat();		//GL_BGRA
+				case gl::INTERNAL_BGRA8_UNORM: return getTranslatedFinalFormat();		//GL_BGRA8_EXT
```

Comment:
you'll have to inver the swizzle for some of these

Reply:
is it for sure necessary? gli reads it and assigns it properly I think, but I do an extra mapping on our API

Reply:
https://github.com/buildaworldnet/IrrlichtBAW/pull/402/files#diff-28a7c4ae6aba8ec71c707eede79b01e9R206 if swizzle of compoment red that will be assigned to it is blue, it will be extra mapped to our format and then assigned to

Reply:
I'd assume that if gli reports `INTERNAL_BGRA8_UNORM` then RGBA = BGRA  and then there's an extra swizzle from gli on top of that

meaning that `INTERNAL_BGRA8_UNORM` needs to become `EF_R8G8B8A8_UNORM` + a swizzle that is

RED = BLUE
GREEN = GREEN
BLUE = RED
ALPHA = ALPHA

Reply:
But that's what my mapping does, it respects it 
Otherwise why does gli read swizzle on 4 size Vector?

Reply:
well but what happens if you have the BGRA format + a custom swizzle? OR will GLI not give you such?

Reply:
the default is `RGBA`, without swizzle assumption
standard swizzle vector with size of 4 also assume that it is `RGBA` and it returns enum swizze to each chanel R as [0], G as [1] and so forth I think. In that case, if my format in case statement matches `BGRA` format, we will have to consider that format is got from texture contating loaded parameters, swizzles though. Mapped swizzles (without mapping extraordinary to our API) are in format order therefore, right? 
For instance:

format[4] (where there is r, g, b, a default)
read swizzle on [0] (`r`), GLI returns on it SWIZZLE_BLUE, so I map it and get 'b'  
read swizzle on [1] (`g`), GLI returns on it SWIZZLE_GREEN, so I map it and get 'g'  
read swizzle on [2] (`b`), GLI returns on it SWIZZLE_RED, so I map it and get 'r'  
read swizzle on [3] (`a`), GLI returns on it SWIZZLE_ALPHA, so I map it and get 'a' 

swizzle output -> `BGRA` 
and there is mapping to our API left

> OR will GLI not give you such?
I have to handle it, right

Reply:
just test and see what assumptions are correct, I know nothing about how GLI works, only our library.

Reply:
status? @Przemog1